lib/api/core/entrypoints/embedded/index.js

```diff
@@ -158,22 +164,30 @@ class EmbeddedEntryPoint extends EntryPoint {
   loadMoreProtocols () {
     const dir = path.join(__dirname, '../../../../../protocols/enabled');
 
-    if (!fs.existsSync(dir)) {
+    let dirs;
+    try {
+      dirs = fs.readdirSync(dir);
+    } catch(e) {
       return Bluebird.resolve();
     }
 
-    const
-      promises = [],
-      dirs = fs.readdirSync(dir)
-        .map(d => path.join(dir, d))
-        .filter(d => fs.statSync(d).isDirectory());
+    dirs = dirs.map(d => path.join(dir, d))
+      .filter(d => fs.statSync(d).isDirectory());
 
-    for (const protoDir of dirs) {
+    return Bluebird.map(dirs, protoDir => {
       const
         protocol = new (require(protoDir))(),
-        name = protocol.protocol || path.basename(protoDir);
+        manifest = new Manifest(this.kuzzle, protoDir, m => {
```

Comment:
Why implement a relatively complex  fallback function mechanism when this is the only place it will ever be used?

Reply:
I had a much simpler way to do it, directly in the manifest class, but protocols need a fallback too, but a different one.
I will simplify that with a different implementation of an abstract Manifest class in the coming batch of fixes.